core/services/functions/connector_handler.go

```diff
@@ -68,6 +70,10 @@ func (h *functionsConnectorHandler) HandleGatewayMessage(ctx context.Context, ga
 		h.lggr.Errorw("request rate-limited", "id", gatewayId, "address", fromAddr)
 		return
 	}
+	if h.subscriptions.GetSubscription(fromAddr) == nil {
```

Comment:
This needs to check against sub balance. So we need required balance in the config and probably a more specific API than GetSubscription().